recipes/open62541/all/test_v1_package/test_package.c

```diff
@@ -0,0 +1,75 @@
+#ifdef UA_ENABLE_AMALGAMATION
+#include <open62541.h>
+#else
+#include <open62541/plugin/log_stdout.h>
+#include <open62541/server.h>
+#include <open62541/server_config_default.h>
+#include <open62541/network_tcp.h>
+#endif
+
+/* Files namespace_foo_flt_generated.h and namespace_foo_flt_generated.c are created from FooFlt.NodeSet2.xml in the
+ * /src_generated directory by CMake */
+#include "foo_flt_nodeids.h"
+#include "namespace_foo_flt_generated.h"
+#include <open62541/plugin/network.h>
+#include <signal.h>
+#include <stdlib.h>
+
+UA_Boolean running = false;
+
+static void server_stop_callback(UA_Server *server, void *data) {
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "stopping server due to timer");
+    running = false;
+}
+
+int main(void) {
+    UA_Server *server = UA_Server_new();
```

Comment:
I'd say that this test_package is too much, and I'd try to reduce it a lot, making it much simpler.
The purpose of test_package is make sure that the package has been created locally. It is not necessary to do a full functional test. Typically showing the version, or calling some auxiliary methods is more than enough.
Also it seems the datasets are very large.

Reply:
Well, I ported the legacy test package code as is, without changing it. 

My original test package was very simple, you can find it [here](https://github.com/Hahn-Schickard/conan-center-index/blob/b011e7bbab6790f7fdc11aadc7140343b99a11fe/recipes/open62541/all/test_package/test_package.cpp), but @Pro, @madebr and @Mo-Tay changed it into the current version in 0e332d885288dcd9d34f48ecb3fdc71ae9bafeee commit, so I would wait for their input before changing the test package again

Reply:
Imho, it's okay to test more than minimally needed, while not going overboard.
But when running the binary, the test should not make any external connections.
So you must make sure to not bind to a port to setup a server.

See e.g. the ulfius test package. In there, the server is built but not run unless you give it a run argument.
that way, we know the ulfius library is built and a user can quickly check that ulfius works by running the test as `./test_package run`.

Reply:
Yes, specifically running anything that enters typical event or dispatch loop like servers is problematic, it can hang CI. The execution should start and finish by itself, as fast as possible, without needing to kill or send any kind of signal. It is also dangerous to do any kind of networking, opening ports or trying any kind of connection, so please avoid that.

Reply:
@memsharded, in that case, I would remove  lines 49 to 71 

```c
 return_code = UA_Server_run(server, &running);

        while(return_code == UA_STATUSCODE_BADCOMMUNICATIONERROR)
        {   
            running = false;
            portNumber = portNumber + 1;
            UA_Server_getConfig(server)->networkLayersSize = 0;
            UA_ServerConfig_addNetworkLayerTCP(UA_Server_getConfig(server), portNumber, 0, 0);
            running = true;
            return_code = UA_Server_run(server, &running);

            if(return_code == UA_STATUSCODE_GOOD)
            {
                printf("free port found");
                break;
            }
            else if(portNumber>4850)
            {
                printf("ports from 4840 to 4850 are not avilable");
                break;
            }
        }
    }
```

This would build the server, log some information and delete the server. That way, the dynamic node generation is still tested, but no server is started, and no `server_stop_callback` would be needed. 

If running the `namespace_foo_flt_generated` is also too much, I would just default back to my original [test_package](https://github.com/Hahn-Schickard/conan-center-index/blob/b011e7bbab6790f7fdc11aadc7140343b99a11fe/recipes/open62541/all/test_package/test_package.cpp) and remove the `server_thread`